src/core/mainloop/connection.c

```diff
@@ -2793,6 +2793,7 @@ retry_listener_ports(smartlist_t *old_conns,
         }
 #ifdef ENABLE_LISTENER_REBIND
         const int may_need_rebind =
+          tor_addr_family(&wanted->addr) == tor_addr_family(&conn->addr) &&
```

Comment:
On the trac ticket you said: ` we check if port matches exactly and *one* of the addresses is a wildcard address:`

Where is the wildcard check happening in this code?
Also, can you please document the previous `may_need_rebind` check and also add the new check?

Reply:
The part that checks for one wildcard address is `bool_neq(tor_addr_is_null(&wanted->addr), tor_addr_is_null(&conn->addr));`. By wildcard address I mean 0.0.0.0 for IPv4 or :: for IPv6.

Wrote a comment in 92bc882ec9440be504f9ebbb714cf6622e455605.